common/src/main/java/whocraft/tardis_refined/common/blockentity/device/ManipulatorCraftingRecipe.java

```diff
@@ -14,11 +14,17 @@ public class ManipulatorCraftingRecipe {
     // List of ingredient blocks for the recipe to work.
     public List<ManipulatorCraftingRecipeItem> itemList;
     // Output to be summoned when the recipe is confirmed.
-    public Item recipeOutput;
+    public Item recipeOutputItem;
+    public Block recipeOutputBlock;
```

Comment:
We should add a boolean flag so that we can specify if the recipe is meant to output a block or item. When we make this recipe data driven we can encode that to JSON and use it much more easily, instead of trying to do null checks all over the place.